homeassistant/components/smarttub/config_flow.py

```diff
@@ -45,9 +52,35 @@ async def async_step_user(self, user_input=None):
             )
 
         existing_entry = await self.async_set_unique_id(account.id)
-        if existing_entry:
-            self.hass.config_entries.async_update_entry(existing_entry, data=user_input)
-            await self.hass.config_entries.async_reload(existing_entry.entry_id)
-            return self.async_abort(reason="reauth_successful")
+        if self._reauth_input is not None:
+            # this is a reauth attempt
+            if existing_entry:
+                if (
+                    existing_entry.unique_id
+                    != self._reauth_input[CONF_CONFIG_ENTRY].unique_id
+                ):
+                    # there is a config entry matching this account, but it is not the one we were trying to reauth
+                    return self.async_abort(reason="already_configured")
+                self.hass.config_entries.async_update_entry(
+                    existing_entry, data=user_input
+                )
+                await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
 
         return self.async_create_entry(title=user_input[CONF_EMAIL], data=user_input)
+
+    async def async_step_reauth(self, user_input=None):
+        """Get new credentials if the current ones don't work anymore."""
+        self._reauth_input = dict(user_input)
+        self._reauth_entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                description_placeholders={"email": self._reauth_input[CONF_EMAIL]},
+                data_schema=DATA_SCHEMA,
+            )
```

Comment:
```suggestion
```

We don't want to switch steps while taking user input 

Then conditionally set `description_placeholders={"email": self._reauth_input[CONF_EMAIL]}` in `async_step_user`

Reply:
We should also assume the user might change their email address as well.

Reply:
The email address from the failed login is displayed to give the user a hint as to which account failed auth, it isn't used otherwise. @frenck was concerned that a user would enter valid credentials for a different account than the one which failed.

Reply:
Is `account.id` the unique value here and not the email address?

If so they may have changed their email address so as long as we validate `account.id` doesn't change (and its not tied to the email) then we should allow them to change their email.

Reply:
Looks like https://github.com/mdz/python-smarttub/blob/main/smarttub/api.py#L137 is the account unique identifier so in theory it should remain stable if the user changes their email address so that is what we need to validate did not change in the reauth process.

Reply:
I copied this code pretty much verbatim from https://developers.home-assistant.io/docs/config_entries_config_flow_handler/, so if this is no longer the preferred approach that should probably be updated

Reply:
I can't say I'm a fan of that approach as I think code that passes user input around blindly in a data structure is riskier, but since there is precedent for it if you want to do it that way it seems ok to me.